code/core/src/main/java/com/adobe/marketing/mobile/internal/DataMarshaller.kt

```diff
@@ -62,26 +63,31 @@ internal object DataMarshaller {
      * @param intent [Intent]
      * @param marshalledData [Map] to add the marshalled data
      */
-    private fun marshalIntentExtras(intent: Intent, marshalledData: MutableMap<String, Any>) {
+    @VisibleForTesting
+    internal fun marshalIntentExtras(intent: Intent, marshalledData: MutableMap<String, Any>) {
         val extraBundle = intent.extras ?: return
-        extraBundle.keySet()?.forEach { key ->
-            val newKey = when (key) {
-                LEGACY_PUSH_MESSAGE_ID -> PUSH_MESSAGE_ID_KEY
-                NOTIFICATION_IDENTIFIER_KEY -> LOCAL_NOTIFICATION_ID_KEY
-                else -> key
-            }
-            try {
-                val value = extraBundle[key]
-                if (value?.toString()?.isNotEmpty() == true) {
-                    marshalledData[newKey] = value
+        try {
+            extraBundle.keySet()?.forEach { key ->
+                val newKey = when (key) {
+                    LEGACY_PUSH_MESSAGE_ID -> PUSH_MESSAGE_ID_KEY
+                    NOTIFICATION_IDENTIFIER_KEY -> LOCAL_NOTIFICATION_ID_KEY
```

Comment:
We can extract our own keys before running the full loop, via direct key access or the Bundle#getString API. This way we can recover the keys we know, then try to read the remaining keys.

Reply:
@emdobrin , @yangyansong-adbe 
Given that the operation that is causing the exception is `extraBundle.keySet()`, may I recommend trying the best case first and then, in-case of an exception try to make a best effort retrieval of known keys as a fallback? That way we treat specific key extraction as a special case and easier code flow, testing.

Reply:
Either way works for me.